[BUG]: `make serve` fails with `ModuleNotFoundError: No module named 'cpex'`

### Summary

`make serve` crashes immediately with `HaltServer: Worker failed to boot.` because `run-gunicorn.sh` activates a **stale** `~/.venv/mcpgateway` that is missing the `cpex>=0.1.0` core dependency added in the 1.0.x release cycle.

---

### Environment

- **OS**: macOS (Darwin 25.3.0, arm64)
- **Branch**: `main`
- **Commit**: `16f8d1a00` (`[Chore] Changes for release of 1.0.1`)

---

### Steps to Reproduce

```bash
make serve
```

---

### Actual Behavior

All gunicorn workers crash on boot:

```
[ERROR] Exception in worker process
Traceback (most recent call last):
  ...
  from cpex.plugins import ToolHookType, ResourceHookType, PromptHookType, ...
ModuleNotFoundError: No module named 'cpex'
[ERROR] Worker (pid:XXXX) exited with code 3.
gunicorn.errors.HaltServer: <HaltServer 'Worker failed to boot.' 3>
```

---

### Expected Behavior

`make serve` should start the production Gunicorn server on `:4444` without errors.

---

### Root Cause

**Two-part failure:**

#### 1. Wrong venv priority in `run-gunicorn.sh` (lines 131–156)

The script resolves the virtual environment in this order:

```
1. ~/.venv/mcpgateway   ← PREFERRED (user global, potentially stale)
2. ${SCRIPT_DIR}/.venv  ← FALLBACK  (project dir, freshly installed by `make install`)
```

This is the **inverse** of what makes sense for a repo-local development workflow. `make serve` calls `make install` first, which installs all packages into the project's `.venv`. Then `run-gunicorn.sh` ignores that freshly-populated venv and picks up the user's global `~/.venv/mcpgateway` instead.

#### 2. `cpex>=0.1.0` added as a new core dependency

`cpex` was added as a mandatory package in `pyproject.toml` (timestamps: `2026-05-11`). Any pre-existing `~/.venv/mcpgateway` created before that date does not have `cpex` and cannot boot the application.

**Affected dependency list** (all missing from stale venv):

```
cpex>=0.1.0
```

---

### Evidence

```bash
# ~/.venv/mcpgateway has gunicorn but NOT cpex:
$ ~/.venv/mcpgateway/bin/pip show cpex
WARNING: Package(s) not found: cpex

# project .venv (created by `make install`) HAS cpex:
$ grep cpex make serve output
 + cpex==0.1.0  ✓ installed

# run-gunicorn.sh activates the wrong venv:
+ source /Users/akshayshinde/.venv/mcpgateway/bin/activate   ← stale venv chosen
```

---

### Impact

`make serve` is **completely broken** for any user who has a pre-existing `~/.venv/mcpgateway` from before the `cpex` dependency was introduced. The `make install` step (which correctly installs `cpex` into the project `.venv`) is silently bypassed.

---

### Fix Options

#### Option A — Reverse venv priority in `run-gunicorn.sh` (recommended)

Change `run-gunicorn.sh` lines 133–142 to check the project `.venv` **first**:

```bash
# Check for virtual environment in script directory (development setup) FIRST
if [[ -f "${SCRIPT_DIR}/.venv/bin/activate" ]]; then
    echo "🔧  Activating virtual environment in script directory"
    source "${SCRIPT_DIR}/.venv/bin/activate"

# Fall back to user home directory venv
elif [[ -f "${HOME}/.venv/mcpgateway/bin/activate" ]]; then
    echo "🔧  Activating virtual environment: ${HOME}/.venv/mcpgateway"
    source "${HOME}/.venv/mcpgateway/bin/activate"
```

This makes `make install` + `make serve` work as a coherent workflow.

#### Option B — Immediate workaround (no code change)

Install `cpex` into the existing global venv:

```bash
~/.venv/mcpgateway/bin/pip install "cpex>=0.1.0"
make serve
```

Or force the project venv explicitly:

```bash
VIRTUAL_ENV=$(pwd)/.venv make serve
```

---

--- a/.secrets.baseline
+++ b/.secrets.baseline
@@ -3,7 +3,7 @@
     "files": "(?x)( package-lock\\.json$ |Cargo\\.lock$ |uv\\.lock$ |go\\.sum$ |mcpgateway/sri_hashes\\.json$ )|^.secrets.baseline$",
     "lines": null
   },
-  "generated_at": "2026-05-13T13:03:20Z",
+  "generated_at": "2026-05-14T12:57:32Z",
   "plugins_used": [
     {
       "name": "AWSKeyDetector"
@@ -5886,7 +5886,7 @@
         "hashed_secret": "64f5490a2808803712ef99d9dfb8bc3ea5a15077",
         "is_secret": false,
         "is_verified": false,
-        "line_number": 296,
+        "line_number": 290,
         "type": "Secret Keyword",
         "verified_result": null
       }

--- a/run-gunicorn.sh
+++ b/run-gunicorn.sh
@@ -125,18 +125,12 @@ echo $$ > "${LOCK_FILE}"
 
 #────────────────────────────────────────────────────────────────────────────────
 # SECTION 3: Virtual Environment Activation
-# Check if a virtual environment is already active. If not, try to activate one
-# from known locations. This ensures dependencies are properly isolated.
+# Check if a virtual environment is already active. If not, activate the one
+# created by `make install` in the project directory.
 #────────────────────────────────────────────────────────────────────────────────
 if [[ -z "${VIRTUAL_ENV:-}" ]]; then
-    # Check for virtual environment in user's home directory (preferred location)
-    if [[ -f "${HOME}/.venv/mcpgateway/bin/activate" ]]; then
-        echo "🔧  Activating virtual environment: ${HOME}/.venv/mcpgateway"
-        # shellcheck disable=SC1090
-        source "${HOME}/.venv/mcpgateway/bin/activate"
-
-    # Check for virtual environment in script directory (development setup)
-    elif [[ -f "${SCRIPT_DIR}/.venv/bin/activate" ]]; then
+    # Use the project-local virtual environment created by `make install`
+    if [[ -f "${SCRIPT_DIR}/.venv/bin/activate" ]]; then
         echo "🔧  Activating virtual environment in script directory"
         # shellcheck disable=SC1090
         source "${SCRIPT_DIR}/.venv/bin/activate"
@@ -146,7 +140,7 @@ if [[ -z "${VIRTUAL_ENV:-}" ]]; then
         echo "⚠️  WARNING: No virtual environment found!"
         echo "   This may lead to dependency conflicts."
         echo "   Consider creating a virtual environment with:"
-        echo "   python3 -m venv ~/.venv/mcpgateway"
+        echo "   make venv install-dev"
 
         # Optional: Uncomment the following lines to enforce virtual environment usage
         # echo "❌  FATAL: Virtual environment required for production deployments"

